File: api/controllers/buscar_saloes.py
from flask import request, jsonify
from database import supabase_admin
from .autenticacao import autenticar

def buscar_saloes():
    token = request.headers.get("Authorization", "").replace("Bearer ", "")
    if not token:
        return jsonify({"sucesso": False, "erro": "Token ausente"}), 401
    
    if not autenticar(token):
        return jsonify({"sucesso": False, "erro": "Token inválido"}), 400

    resposta = supabase_admin.table("saloes").select("*").execute()

    if not resposta.data:
        return jsonify({"sucesso": False, "erro": "Falha ao encontrar salões"}), 500
        
    return jsonify({"sucesso": True, "saloes": resposta.data})

File: web/src/pages/Agendamento/services/buscarSaloes.js
import { pegarSessao } from "../../../services/pegarSessao.js"

export async function buscarSaloes() {
    try {
        const token = await pegarSessao()

        if (!token) {
            return false
        }

        const resposta = await fetch("/api/buscar/saloes", {
            method: "GET",
            headers: {
                'Authorization': `Bearer ${token.access_token}`
            }
        })

        const resultado = await resposta.json()

        if (!resultado.sucesso) {
            console.log(resultado.erro)
            return false
        }

        return resultado.saloes

    } catch (erro) {
        console.log("Erro ao buscar salões:", erro)
        return false
    }

}
